plugin/trino-spooling-filesystem/src/test/java/io/trino/spooling/filesystem/TestFileSystemSegmentPruner.java

```diff
@@ -41,110 +42,121 @@
 @TestInstance(PER_CLASS)
 class TestFileSystemSegmentPruner
 {
-    private static final String TEST_LOCATION = "memory://";
+    private static final List<FileSystemLayout> LAYOUTS = ImmutableList.of(new SimpleFileSystemLayout(), new PartitionedFileSystemLayout(new PartitionedLayoutConfig()
+            .setPartitions(8)));
+
+    private static final Location TEST_LOCATION = Location.of("memory://");
 
     private static final FileSystemSpoolingConfig SPOOLING_CONFIG = new FileSystemSpoolingConfig()
-            .setLocation(TEST_LOCATION)
+            .setLocation(TEST_LOCATION.toString())
             .setPruningBatchSize(1);
 
     @Test
     public void shouldPruneExpiredSegments()
     {
-        MemoryFileSystem fileSystem = new MemoryFileSystem();
-        try (ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor()) {
-            FileSystemSegmentPruner pruner = new FileSystemSegmentPruner(SPOOLING_CONFIG, _ -> fileSystem, executorService);
+        for (FileSystemLayout layout : LAYOUTS) {
```

Comment:
Create specific methods for each layout type.
Or if you prefer make this test abstract and create subclass per layout.
For loops in tests suck - and for some reason we do not want to have providers in code (I mean - I want - but not everyone)